state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -2147,8 +2156,79 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		origin: DepositOrigin<T, I>,
 	) -> DepositAction<T, I> {
 		match action.clone() {
-			ChannelAction::LiquidityProvision { lp_account, .. } => {
-				T::Balance::credit_account(&lp_account, asset.into(), amount_after_fees.into());
+			ChannelAction::LiquidityProvision { lp_account, additional_action, refund_address } => {
+				let used_for_flip_funding_swap = match additional_action {
+					Some(AdditionalDepositAction::FundFlip { flip_amount_to_credit })
+						if !frame_system::Pallet::<T>::account_exists(&lp_account) =>
+					{
+						let is_flip_asset = matches!(asset.into(), Asset::Flip);
+						let funding_amount = if is_flip_asset {
+							amount_after_fees.into()
+						} else {
+							INITIAL_FLIP_FUNDING
+						};
+
```

Comment:
So if the deposit asset is FLIP, we transfer all of it to the native balance? 

This sort of makes sense, but also might be surprising... I think I'd prefer the same behavioud for any asset. 
Even if it's FLIP, it's unlikely that an LP would want a large amount of FLIP in their native balance.